File: core/src/bara/game/combat/BulletDef.java
package bara.game.combat;

import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.utils.Array;
import bara.game.behavior.Behavior;

public class BulletDef {
    public Vector2 velocity;
    public Vector2 position;
    public Behavior behavior;
    public Array<Effect> effects;

    public BulletDef() {
        velocity = new Vector2();
        position = new Vector2();
        behavior = null;
        effects = new Array<Effect>();
    }
}

File: core/src/bara/game/combat/BulletDef.kt
package bara.game.combat

import com.badlogic.gdx.math.Vector2
import bara.game.behavior.Behavior
import ktx.collections.GdxArray

class BulletDef(
    val velocity: Vector2 = Vector2(),
    val position: Vector2 = Vector2(),
    val behavior: Behavior? = null,
    val effects: GdxArray<Effect> = GdxArray()
) {

}
